Automattic/wp-calypso
# Site Migration: Add assistance modal for migration

Related to pdDR7T-1uw-p2

## Proposed Changes

* This is an initial task part of the migration flow migration plan pdDR7T-1uw-p2, this should be behind a feature flag (`migration_assistance_modal`) and for now, invisible to all users.

This PR adds a modal asking users to subscribe to the creator plan and get help from HE on the site migration. The modal should be triggered when the user navigates back using the stepper back button, and should only be visible on flows that are presenting this plan step.
<img width="1022" alt="image" src="https://github.com/Automattic/wp-calypso/assets/2653810/f84ed15c-3288-4699-9f34-70b8cb6b47ce">


According to this pcmemI-2sX-p2 great post, I have mapped some flows using this step and furthermore should show the modal when navigating back, they are:

 * site-setup
 * site-migration-flow
 * import-hosted-sites 
 * import-flow

## Testing Instructions




Take this post (pcmemI-2sX-p2) as a reference to all migration entry points, we should test all entry points to make sure the modal only appears on the appropriate flow.

The appropriate flows are the ones using the plan step pasted above.

* Using live link or pull and run this branch
* Navigate to the flows, adapting the URL with live link or calypso.localhost.
* Make sure to add the feature flag in the URL to be able to see the modal when navigating back `flags=migration_assistance_modal`

The modal loos like this, and copies are not final yet:
<img width="797" alt="image" src="https://github.com/Automattic/wp-calypso/assets/2653810/724d0566-c9d0-4151-aed3-8278c2c89234">


## Pre-merge Checklist
